### PR TITLE
Ext/GitHub smart notification

### PR DESCRIPTION
**Pull request by shoppingjaws**
## Description

Github Notification Viewer
- automatically read unnecessary notification
- automatically read notification based on configured rule

## Screencast
github-smart-notification 2024-05-05 at 17 33 09

github-smart-notification 2024-05-05 at 17 33 23

github-smart-notification 2024-05-05 at 17 34 07

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/github-smart-notification/src/view/configurations.view.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                title="See Detail"
+                target={
+                  <Detail
+                    markdown={`# ${s.description}\n**Repository** = ${s.repository}\n\n**Title** = ${s.title}\n\n**Reason** = ${s.reason}\n\n
+              `}
+                  />
+                }
+              />
+              <Action
+                title="Delete"
```
Comment (pernielsentikaer) [inline extensions/github-smart-notification/src/view/configurations.view.tsx:27]:
```suggestion
                title="Delete"
                style={Action.Style.Destructive}
                shortcut={Keyboard.Shortcut.Common.Remove}
```

File: extensions/github-smart-notification/src/view/configurations.view.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { Action, ActionPanel, Detail, List, environment } from "@raycast/api";
```
Comment (pernielsentikaer) [inline extensions/github-smart-notification/src/view/configurations.view.tsx:1]:
```suggestion
import { Action, ActionPanel, Detail, List, environment, Keyboard } from "@raycast/api";
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- What about adding some defaults path and set gh path to `required: false` instead?

- Add so it's possible to add a new configuration directly from `List Configurations` 

- What about making adding a `no-view` command that will run every 10 minutes automatically

- Maybe include icons and color when creating a new notification, maybe soemthing like this

Raycast10052024-0YBo8svv png@2x



I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by shoppingjaws** [comment]
# changes

Thank you for reviewing. re-submit my PR.

- move form view from root command into configuration view.
- set default command path. 
- set notification view item icons sourced from repository icon

**Comment by shoppingjaws** [comment]
@pernielsentikaer  Updated 👍

File: extensions/github-smart-notification/package.json [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    },
+    {
+      "name": "autoReadMergedPr",
+      "title": "automatically read merged PR",
+      "description": "if true, merged PR will be read",
+      "default": true,
+      "type": "checkbox",
+      "label": "checkbox",
+      "required": true
+    }
```
Comment (pernielsentikaer) [inline extensions/github-smart-notification/package.json:84]:
```suggestion
    {
      "name": "ghCommandPath",
      "title": "gh Command Binary PATH",
      "description": "Installed Path of gh command",
      "type": "textfield",
      "required": false,
      "default": "/opt/homebrew/bin/gh"
    },
    {
      "name": "autoReadMergedPr",
      "title": "Options",
      "description": "if true, merged PR will be read",
      "default": true,
      "type": "checkbox",
      "label": "Automatically mark merged PRs as read",
      "required": true
    }
```

File: extensions/github-smart-notification/src/view/notifications.view.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+          ]}
+          actions={
+            <ActionPanel>
+              <Action
+                title="Open"
+                onAction={() => toHtmlUrl(s).then((url) => execAsync(`open ${url}`))}
+                shortcut={Keyboard.Shortcut.Common.Open}
+              />
+              <Action
+                title="Mark As Read"
```
Comment (pernielsentikaer) [inline extensions/github-smart-notification/src/view/notifications.view.tsx:61]:
```suggestion
                title="Mark as Read"
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

I have now tested your extension, and I have some feedback ready for you:

- If you're opening the menubar command as the first thing, then there is no data since it doesn't fetch items, maybe it should do that?

Raycast04062024-ZBKQpQxe png@2x

- It's still possible to add a new configuration without a description

github-smart-notification 2024-06-04 at 10 04 33

github-smart-notification 2024-06-04 at 10 05 11

- add a [confirmAlert](https://developers.raycast.com/api-reference/feedback/alert#confirmalert) when you delete the notification and add a missing icon `Icon.Trash` maybe?

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by shoppingjaws** [comment]
@pernielsentikaer Hi, I updated.

- add alert and trash icon to the configuration view on delete.

> If you're opening the menubar command as the first thing, then there is no data since it doesn't fetch items, maybe it should do that?
- You mean adding the loading logic, right?

**Comment by pernielsentikaer** [comment]
Yes, that would make sense since the command and menubar uses the same cache

**Comment by shoppingjaws** [comment]
@pernielsentikaer Ive already added loading logic to the menubar.tsx.
when isLoading is set true when the state is undefined. The menubar becomes invisible while the fetching task is in progress. Isn’t this enough?

**Comment by shoppingjaws** [comment]
@pernielsentikaer yes, please 👍

**Comment by mil3na** [comment]
Hi! Per is on vacation, so I will be reviewing your extension.
I wonder if this wouldn't be a better addition to the Github extension instead. https://www.raycast.com/raycast/github/commands 
There, we already have a notification command, you could add a filter as a preference.
What do you think?

**Comment by shoppingjaws** [comment]
@pernielsentikaer @mil3na That's good idea. Is it too late to suggest that?
I'd love to port this feature to a Github extension, but why not wait until you've released it?

**Comment by mil3na** [comment]
The Github extension is opensource and open for contributions. Feel free to add this functionality there! I will close this PR for now :)

**Comment by shoppingjaws** [comment]
If so, I wish you had suggested it a little earlier.

**Pull request closed without merging**